### ISSUE TITLE
Multiple definitions linker issue

### ISSUE THREAD

**Issue by voivoid** [boostorg/safe_numerics#71]
Hi!

I see a linker error while I try to use safe_numerics in at least two translation units:
```
/usr/bin/ld: ../problems/CMakeFiles/AocProblems.dir/src/2017/problem_18.cpp.o: in function `boost::safe_numerics::make_error_code(boost::safe_numerics::safe_numerics_error const&)':
problem_18.cpp:(.text+0x0): multiple definition of `boost::safe_numerics::make_error_code(boost::safe_numerics::safe_numerics_error const&)'; ../problems/CMakeFiles/AocProblems.dir/src/2017/problem_07.cpp.o:problem_07.cpp:(.text+0x0): first defined here
```

Declaring make_error_code function as an inline fixes the problem

Tested with clang 8.0 / gcc 8.3.0 and boost 1.70

**Comment by robertramey** [comment]
`
// constexpr - damn, can't use constexpr due to std::error_code
std::error_code make_error_code(const safe_numerics_error & e){
    return std::error_code(static_cast<int>(e), safe_numerics_error_category);
}
`
Looks like a good call.  Note the comment I placed here.  I'll make the change on my development system and let it migrate to the next boost version.

Thanks for figuring this out and letting me know.

Robert Ramey

**Comment by voivoid** [comment]
Fixed in boost 1.71

Thanks!
